js/components/dataset/restore-modal.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+<div>
+<modal :title="_('Confirm restore')"
+    class="modal-info dataset-delete-modal"
+    v-ref:modal>
+
+    <div class="modal-body">
+        <p class="lead text-center">
+            {{ _('You are about to restore this dataset') }}
+        </p>
+        <p class="lead text-center">
+            {{ _('Are you sure?') }}
+        </p>
+    </div>
+
+    <footer class="modal-footer text-center">
+        <button type="button" class="btn btn-success btn-flat pull-left"
+            @click="confirm">
+            {{ _('Confirm') }}
+        </button>
+        <button type="button" class="btn btn-warning btn-flat"
+            @click="$refs.modal.close">
+            {{ _('Cancel') }}
+        </button>
+    </footer>
+</modal>
+</div>
+</template>
+
+<script>
+import API from 'api';
+import Modal from 'components/modal.vue';
+
+export default {
+    components: {Modal},
+    props: {
+        dataset: Object,
+    },
+    methods: {
+        confirm() {
+            var restore_dataset =  this.dataset;
```

Comment:
Use `const` instead of `var` because we are trying to use ES6 as much as possible